src/components/comment-section/CommentSection.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useState, useEffect } from 'react';
+import { Comment } from '../../types/types';
+import './CommentSection.css';
+
+type CommentSectionProps = {
+  currentUserId: number;
+  currentUserName: string;
+  initialComments: Comment[] | undefined;
+  onCommentsChange: (comments: Comment[]) => void;
+};
+
+const CommentSection: React.FC<CommentSectionProps> = ({
+  currentUserId,
+  currentUserName,
+  initialComments,
+  onCommentsChange,
+}) => {
+  const [comments, setComments] = useState<Comment[] | undefined>(
+    initialComments,
+  );
+  const [showCommentInput, setShowCommentInput] = useState(false);
+  const [newComment, setNewComment] = useState('');
+
+  useEffect(() => {
```

Comment:
this useEffect is not needed as well, since you are getting comments from parent component